File: src/python_nodes/dataHandler.py
#!/home/mhyde/vEnvs/rosPy/bin/python

'''
ROS Service - Client:

Requests coordinates from the CSV File
to the Data Distributer upon service request.

'''

import sys
import rospy
from motor_control.srv import *

def requestCoordinate(request):
    rospy.wait_for_service('coordinatePass')
    try:
        srvReq = rospy.ServiceProxy('coordinatePass', coordinatePass)
        resp = srvReq(request)
        return resp
    except rospy.ServiceException, e:
        rospy.loginfo('Service Call Failed: %s'%e)

def usage():
    return '%s [req]'%sys.argv[0]

if __name__ == '__main__':
    if len(sys.argv) == 1:
        req = True
    else:
        print usage()
    print "Requesting Coordinate ... "
    coor = requestCoordinate(req)
    print '[%s, %s, %s]'%(coor.xCoordinate, coor.yCoordinate, coor.zCoordinate)
    

File: src/cpp_nodes/dataHandler.cpp
#include "ros/ros.h"
#include "motor_control/coordinatePass.h"
#include "dataHandler.h"

DataHandle::DataHandle(){

}

void DataHandle::requestCoordinate(){

  ros::NodeHandle node;
  ros::ServiceClient client = node.serviceClient<motor_control::coordinatePass>("coordinatePass");
  motor_control::coordinatePass srv;
  srv.request.request = true;
  client.call(srv);

  std::cout << srv.response.xCoordinate << std::endl;
  std::cout << srv.response.yCoordinate << std::endl;
  std::cout << srv.response.zCoordinate << std::endl;
}

DataHandle::~DataHandle(){
}

//
//  #           NOTE:             #
//  #   TESTING CLASS FUNCTION    #
//  # SCRIPT SHOULD NOT CALL MAIN #
//

int main(int argc, char **argv){
  
  ros::init(argc, argv, "coordinatePassClient");
  DataHandle client;

  client.requestCoordinate();

  return 0;
}